src/Microsoft.Azure.WebJobs.Host/Queues/Listeners/QueueListener.cs

```diff
@@ -244,13 +244,23 @@ private TaskSeriesCommandResult CreateSucceededResult()
 
         internal async Task ProcessMessageAsync(IStorageQueueMessage message, TimeSpan visibilityTimeout, CancellationToken cancellationToken)
         {
+            // These values may change if the message is inserted into another queue. We'll store them here and make sure
+            // the message always has the original values before we pass it to a customer-facing method.
+            string id = message.Id;
+            string popReceipt = message.PopReceipt;
+            DateTimeOffset? insertionTime = message.InsertionTime;
+            DateTimeOffset? nextVisibleTime = message.NextVisibleTime;
+            DateTimeOffset? expirationTime = message.ExpirationTime;
+
             try
             {
                 if (!await _queueProcessor.BeginProcessingMessageAsync(message.SdkObject, cancellationToken))
                 {
                     return;
                 }
 
+                message.SdkObject.SetProperties(id, popReceipt, insertionTime, nextVisibleTime, expirationTime);
```

Comment:
It was difference between 1 and 3 updates, and I decided to go for completeness. I don't know what people may be doing from within their custom QueueProcessors and I'd rather not break things when we have a way to keep the existing behavior with 2 extra lines of code. I find it hard to explain "your Id and PopReceipt may have changed in these calls, but we'll reset it in this one".